types/module/module.go

```diff
@@ -391,7 +391,9 @@ func (m Manager) RunMigrations(ctx sdk.Context, cfg Configurator, fromVM Version
 	}
 
 	updatedVM := make(VersionMap)
-	for moduleName, module := range m.Modules {
+
+	for _, moduleName := range m.OrderInitGenesis {
+		module := m.Modules[moduleName]
```

Comment:
maybe we should return an error if module is nil?

Reply:
the assumption is "orderinitgenesis" should contain all modules, I think? The more common mistake tends to be someone omits putting a moduleName there -- we can possibly add a few error checks there

Reply:
I updated it to just use sorted module names

Reply:
> the assumption is "orderinitgenesis" should contain all modules, I think?

OK, we can make this assumption.

I actually think it's more flexible to use orderInitGenesis here? There might be some cases where the chain developer wants moduleB to be migrated before moduleA, wdyt?

Reply:
possible -- not sure if orderInitGenesis will always be the desired migration order, but makes sense. I added extra checks there